Cython/Compiler/Pythran.py

```diff
@@ -0,0 +1,154 @@
+from .PyrexTypes import BufferType, CType, CTypedefType, CStructOrUnionType
+
+_pythran_var_prefix = "__pythran__"
+# Pythran/Numpy specific operations
+def has_np_pythran(env):
+    scope = env.outer_scope
+    while scope != None:
+        scope = scope.outer_scope
+    ctx = env.global_scope().context
+    if ctx is None:
+        return False
+    options = ctx.options
+    return options is not None and options.np_pythran
+
+def is_pythran_supported_dtype(type_):
+    if isinstance(type_, CTypedefType):
+        return is_pythran_supported_type(type_.typedef_base_type)
+    return hasattr(type_,"is_numeric") and type_.is_numeric
+
+def pythran_type(Ty,ptype="ndarray"):
+    if Ty.is_buffer:
+        ndim,dtype = Ty.ndim, Ty.dtype
+        if isinstance(dtype, CStructOrUnionType):
+            ctype = dtype.cname
+        elif isinstance(dtype, CType):
+            ctype = dtype.sign_and_name()
+        elif isinstance(dtype, CTypedefType):
+            ctype = dtype.typedef_cname
+        else:
+            raise ValueError("unsupported type %s!" % str(dtype))
+        return "pythonic::types::%s<%s,%d>" % (ptype,ctype, ndim)
+    from .PyrexTypes import PythranExpr
+    if Ty.is_pythran_expr:
+        return Ty.pythran_type
+    if hasattr(Ty,"is_none") and Ty.is_none:
+        return "decltype(pythonic::__builtin__::None)"
+    if Ty.is_numeric:
+        return Ty.sign_and_name()
+    raise ValueError("unsupported pythran type %s (%s)" % (str(Ty), str(type(Ty))))
+    return None
+
+def type_remove_ref(ty):
+    return "typename std::remove_reference<%s>::type" % ty
+
+def pythran_binop_type(op, tA, tB):
+    return "decltype(std::declval<%s>() %s std::declval<%s>())" % \
+        (pythran_type(tA), op, pythran_type(tB))
+
+def pythran_unaryop_type(op, type_):
+    return "decltype(%sstd::declval<%s>())" % \
+        (op, pythran_type(type_))
```

Comment:
Fixed in https://github.com/cython/cython/pull/1607/commits/e3fb29c91b13edcc9297b02b0ca10e9a3da3928d